### DISCUSSION TITLE
How to generate and set avatar?

### DISCUSSION THREAD

**Discussion by Dirk-Ingenieur** [bbernhard/signal-cli-rest-api#540] [Q&A]
I successfully managed to set a profile NAME - now I am curious how to set an avatar image.

As far as I understood one could either choose a predefined avatar or upload a picture.
In any case I need the base64_string to be passed with the curl command.

Could anybody help which strings are predefined or how I might even encode and upload a picture?

**Comment by bbernhard** [comment]
It should work like this:

```
curl -X 'PUT' \
  'http://127.0.0.1:8080/v1/profiles/<your signal number>' \
  -H 'accept: application/json' \
  -H 'Content-Type: application/json' \
  -d '{
  "base64_avatar": "<the base64 encoded profile>",
  "name": "<your username>"
}'
```

To base64 encode an image you can e.g use the `base64` commandline tool. e.g: `base64 /path/to/your/image.jpg`

**Reply by Dirk-Ingenieur** [reply]
This is actually what I did, but whenever I paste the long string instead of <the base64 encoded profile> I receive an error:
{"error":"Couldn't process request - invalid request"}

**Comment by FerNerd** [comment]
Works fine for me, but you have to use base64 as one string with so `base64 -w0 /path/to/your/image.jpg`.
